tests/azure_iot_hub_client_ut.c

```diff
@@ -0,0 +1,351 @@
+                          eAzureIoTHubClientSuccess );
+
+    /* Fail init when null Transport passed */
+    assert_int_not_equal( AzureIoTHubClient_Init( &xTestIoTHubClient,
+                                                  ucHostname, sizeof( ucHostname ) - 1,
+                                                  ucDeviceId, sizeof( ucDeviceId ) - 1,
+                                                  &xHubClientOptions,
+                                                  ucBuffer, sizeof( ucBuffer ),
+                                                  prvGetUnixTime, NULL ),
+                          eAzureIoTHubClientSuccess );
+
+    /* Fail when smaller buffer is passed */
+    assert_int_not_equal( AzureIoTHubClient_Init( &xTestIoTHubClient,
+                                                  ucHostname, sizeof( ucHostname ) - 1,
+                                                  ucDeviceId, sizeof( ucDeviceId ) - 1,
+                                                  &xHubClientOptions,
+                                                  ucBuffer, azureiotUSERNAME_MAX,
+                                                  prvGetUnixTime, &xTransportInterface ),
+                          eAzureIoTHubClientSuccess );
+
+    /* Fail init when AzureIoTMQTT_Init fails */
+    will_return( AzureIoTMQTT_Init, eAzureIoTMQTTNoMemory );
+    assert_int_not_equal( AzureIoTHubClient_Init( &xTestIoTHubClient,
+                                                  ucHostname, sizeof( ucHostname ) - 1,
+                                                  ucDeviceId, sizeof( ucDeviceId ) - 1,
+                                                  &xHubClientOptions,
+                                                  ucBuffer, sizeof( ucBuffer ),
+                                                  prvGetUnixTime, &xTransportInterface ),
+                          eAzureIoTHubClientSuccess );
+}
+
+static void testAzureIoTHubClient_Init_Success( void ** state )
+{
+    AzureIoTHubClient_t xTestIoTHubClient;
+    AzureIoTHubClientOptions_t xHubClientOptions = { 0 };
+
+    ( void ) state;
+    will_return( AzureIoTMQTT_Init, eAzureIoTMQTTSuccess );
+
+    assert_int_equal( AzureIoTHubClient_Init( &xTestIoTHubClient,
+                                              ucHostname, sizeof( ucHostname ) - 1,
+                                              ucDeviceId, sizeof( ucDeviceId ) - 1,
+                                              &xHubClientOptions,
+                                              ucBuffer,
+                                              sizeof( ucBuffer ),
+                                              prvGetUnixTime,
+                                              &xTransportInterface ),
+                      eAzureIoTHubClientSuccess );
+}
+
+static void testAzureIoTHubClient_Deinit_Success( void ** state )
+{
+    AzureIoTHubClient_t xTestIoTHubClient;
+
+    ( void ) state;
+
+    AzureIoTHubClient_Deinit( &xTestIoTHubClient );
```

Comment:
Can we init and then do deinit?
Something like this:
{
    AzureIoTHubClient_t xTestIoTHubClient;

    ( void ) state;

 prvSetupTestIoTHubClient( &xTestIoTHubClient );

AzureIoTHubClient_Deinit( &xTestIoTHubClient );
}